sklearn/utils/estimator_checks.py

```diff
@@ -1197,6 +1198,46 @@ def check_classifiers_one_label(name, classifier_orig):
             raise exc
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_classifiers_one_label_sample_weights(name, classifier_orig):
+    # check that classifiers accepting sample_weight fit fine or
+    # throws an ValueError if the problem is reduce to one class.
+    error_fit = ("Classifier can't train when only one class is present "
+                 "after sample_weight trimming.")
+    error_predict = ("Classifier can't predict when only one class is "
+                     "present after sample_weight trimming.")
+    if has_fit_parameter(classifier_orig, "sample_weight"):
```

Comment:
has_fit_parameter will return false if **kwargs are accepted. I would just try fit with this kwarg and continue if a TypeError is raised